pkg/proxy/iptables/proxier.go

```diff
@@ -682,11 +713,16 @@ func (proxier *Proxier) OnServiceCIDRsChanged(_ []string) {}
 // then encoding to base32 and truncating to 16 chars. We do this because IPTables
 // Chain Names must be <= 28 chars long, and the longer they are the harder they are to read.
 func portProtoHash(servicePortName string, protocol string) string {
-	hash := sha256.Sum256([]byte(servicePortName + protocol))
-	encoded := base32.StdEncoding.EncodeToString(hash[:])
+	encoded := base32SHA256Sum(servicePortName, protocol)
 	return encoded[:16]
```

Comment:
since this also has a max length, you could make `base32SHA256Sum` take a `maxLength` arg